### ISSUE TITLE
ci instability: /loki/api/v1/tail test: ERR_ASSERTION

### ISSUE THREAD

**Issue by jgehrcke** [opstrace/opstrace#1372]
```
[2021-09-07T06:46:52Z]   1) Loki tail API test suite
[2021-09-07T06:46:52Z]        connect and stream cortex ingester pod logs from /loki/api/v1/tail:
[2021-09-07T06:46:52Z]
[2021-09-07T06:46:52Z]       AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
[2021-09-07T06:46:52Z] + actual - expected
[2021-09-07T06:46:52Z]
[2021-09-07T06:46:52Z] + 'timeout'
[2021-09-07T06:46:52Z] - 'test'
[2021-09-07T06:46:52Z]       + expected - actual
[2021-09-07T06:46:52Z]
[2021-09-07T06:46:52Z]       -timeout
[2021-09-07T06:46:52Z]       +test
[2021-09-07T06:46:52Z]
[2021-09-07T06:46:52Z]       at Context.<anonymous> (test_loki_tail_endpoint.ts:84:12)
```
https://buildkite.com/opstrace/scheduled-main-builds/builds/3053#f98e2b69-00d1-4157-b2cb-d814677c35c8/808-2727

**Comment by triclambert** [comment]
```
[2021-09-07T15:52:11Z]   20 passing (7m)
[2021-09-07T15:52:11Z]   4 pending
[2021-09-07T15:52:11Z]   3 failing
[2021-09-07T15:52:11Z]
[2021-09-07T15:52:11Z]   1) Loki tail API test suite
[2021-09-07T15:52:11Z]        connect and stream cortex ingester pod logs from /loki/api/v1/tail:
[2021-09-07T15:52:11Z]
[2021-09-07T15:52:11Z]       AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
[2021-09-07T15:52:11Z] + actual - expected
[2021-09-07T15:52:11Z]
[2021-09-07T15:52:11Z] + 'timeout'
[2021-09-07T15:52:11Z] - 'test'
[2021-09-07T15:52:11Z]       + expected - actual
[2021-09-07T15:52:11Z]
[2021-09-07T15:52:11Z]       -timeout
[2021-09-07T15:52:11Z]       +test
[2021-09-07T15:52:11Z]
[2021-09-07T15:52:11Z]       at Context.<anonymous> (test_loki_tail_endpoint.ts:84:12)
[2021-09-07T15:52:11Z] 
[2021-09-07T15:52:11Z]   2) system logs test suite
[2021-09-07T15:52:11Z]        check loki ingester logs:
[2021-09-07T15:52:11Z]      Error: Expectation not fulfilled within 30 s
[2021-09-07T15:52:11Z]       at waitForLokiQueryResult (testutils/logs.ts:266:9)
[2021-09-07T15:52:11Z]       at runMicrotasks (<anonymous>)
[2021-09-07T15:52:11Z]       at processTicksAndRejections (node:internal/process/task_queues:96:5)
[2021-09-07T15:52:11Z]       at async query (test_systemlogs.ts:52:20)
[2021-09-07T15:52:11Z]       at async Context.<anonymous> (test_systemlogs.ts:66:5)
[2021-09-07T15:52:11Z] 
[2021-09-07T15:52:11Z]   3) system logs test suite
[2021-09-07T15:52:11Z]        check cortex distributor logs:
[2021-09-07T15:52:11Z]      Error: Expectation not fulfilled within 30 s
[2021-09-07T15:52:11Z]       at waitForLokiQueryResult (testutils/logs.ts:266:9)
[2021-09-07T15:52:11Z]       at runMicrotasks (<anonymous>)
[2021-09-07T15:52:11Z]       at processTicksAndRejections (node:internal/process/task_queues:96:5)
[2021-09-07T15:52:11Z]       at async query (test_systemlogs.ts:52:20)
[2021-09-07T15:52:11Z]       at async Context.<anonymous> (test_systemlogs.ts:72:5)
```

https://buildkite.com/opstrace/scheduled-main-builds/builds/3056#60ca7d72-514c-41a7-81cc-7b3cf0181fec/6-1852

**Comment by nickbp** [comment]
seen in a PR build https://buildkite.com/opstrace/prs/builds/5748#0304179c-6ad9-48b0-a090-4e7ebadbb3bb/826-2750

**Comment by jgehrcke** [comment]
> however this lacks the assert error, only has the "expectation not fulfilled" errors:

@nickbp the relevant pattern in 'your' log excerpt is
```
[2021-09-07T22:47:25Z] 2021-09-07T22:47:25.332Z info: globalTestSuiteSetupOnce()
[2021-09-07T22:47:25Z] 2021-09-07T22:47:25.333Z info: waitForLokiQueryResult() loop: attempt query 1
[2021-09-07T22:47:35Z] 2021-09-07T22:47:35.335Z info: waitForLokiQueryResult() loop: http request failed: Timeout awaiting 'request' for 10000ms -- ignore, proceed with next iteration
[2021-09-07T22:47:36Z] 2021-09-07T22:47:36.337Z info: waitForLokiQueryResult() loop: attempt query 2
[2021-09-07T22:47:46Z] 2021-09-07T22:47:46.344Z info: waitForLokiQueryResult() loop: http request failed: Timeout awaiting 'request' for 10000ms -- ignore, proceed with next iteration
[2021-09-07T22:47:47Z] 2021-09-07T22:47:47.345Z info: waitForLokiQueryResult() loop: attempt query 3
[2021-09-07T22:47:57Z] 2021-09-07T22:47:57.353Z info: waitForLokiQueryResult() loop: http request failed: Timeout awaiting 'request' for 10000ms -- ignore, proceed with next iteration
[2021-09-07T22:47:57Z] 2021-09-07T22:47:57.353Z error: query deadline hit ()
[2021-09-07T22:47:57Z]     1) check loki ingester logs
```
which is https://github.com/opstrace/opstrace/issues/1226 -- I carried this over: https://github.com/opstrace/opstrace/issues/1226#issuecomment-915185855

**Comment by jgehrcke** [comment]
```
[2021-09-07T18:45:03Z]   1) Loki tail API test suite
[2021-09-07T18:45:03Z]        connect and stream cortex ingester pod logs from /loki/api/v1/tail:
[2021-09-07T18:45:03Z]
[2021-09-07T18:45:03Z]       AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
[2021-09-07T18:45:03Z] + actual - expected
[2021-09-07T18:45:03Z]
[2021-09-07T18:45:03Z] + 'timeout'
[2021-09-07T18:45:03Z] - 'test'
[2021-09-07T18:45:03Z]       + expected - actual
[2021-09-07T18:45:03Z]
[2021-09-07T18:45:03Z]       -timeout
[2021-09-07T18:45:03Z]       +test
```
https://buildkite.com/opstrace/prs/builds/5747#915e80bc-b26f-499f-a3ad-2bed846028c6/364-369

**Comment by nickbp** [comment]
I don't know if this has been mentioned already, but this failure seems to be pretty tightly focused on the upgrade tests. Scheduled main tests are having their own occasional failures but they seem to be unrelated (e.g. DNS issues, cert issues). Meanwhile upgrade tests are failing pretty often and always with this issue.

https://buildkite.com/opstrace/upgrade-tests/builds/1366#6f03db90-3d3f-4833-abed-9f979535f56d/2408-4326
https://buildkite.com/opstrace/upgrade-tests/builds/1368#b542cc03-f9d7-4b16-b3a9-a31dcc1ea1c1/6-1614
https://buildkite.com/opstrace/upgrade-tests/builds/1369#45cb7d5f-7b42-46dd-8c29-dc6934be17a3/634-2563
https://buildkite.com/opstrace/upgrade-tests/builds/1370#1d446aee-e98d-411e-b8d4-ba1ca71ccc1a/2642-4575
https://buildkite.com/opstrace/upgrade-tests/builds/1372#7d5cf7fa-97a3-4046-bee0-35a54ea100af/993-2905
https://buildkite.com/opstrace/upgrade-tests/builds/1373#a99f59a6-f341-4a66-aada-84b472dc900e/2452-4367 <- also had an unrelated GCP failure
https://buildkite.com/opstrace/upgrade-tests/builds/1374#eb3afa1a-374d-4bd7-8823-09d1c3c1698a/279-2216
https://buildkite.com/opstrace/upgrade-tests/builds/1376#e44ead33-453c-4599-b35d-6b2f8ec09f00/656-2588
https://buildkite.com/opstrace/upgrade-tests/builds/1377#661f50e7-b09f-4b63-9d48-a9dde9f1b3b4/30-1955
https://buildkite.com/opstrace/upgrade-tests/builds/1378#3097eb89-e096-4527-91ff-b6c55ef97004/446-2369
https://buildkite.com/opstrace/upgrade-tests/builds/1380#39606f4e-e946-47f0-8ec6-887c09f1ad3c/651-2574

**Comment by nickbp** [comment]
https://buildkite.com/opstrace/upgrade-tests/builds/1381#882e8387-639b-4851-a6f2-cfdc1cab017f/266-2204

**Comment by sreis** [comment]
> I don't know if this has been mentioned already, but this failure seems to be pretty tightly focused on the upgrade tests.

@nickbp This is related to #1226 (the fix we have doesn't handle upgrades) should be fixed when we close #1383.
